### DISCUSSION TITLE
Multiple designs for each component

### DISCUSSION THREAD

**Discussion by Ari24-cb24** [aridevelopment-de/myanimetab#144] [Brainstorming]
Somehow an option to select between visual styles of components. Take for example the clock. This is the current state:
![grafik](https://user-images.githubusercontent.com/56089155/210894515-a36b564d-b46b-40bb-a754-7d91023575a5.png)

Would be nice if the user is able to change it to something like this:
![grafik](https://user-images.githubusercontent.com/56089155/210894564-d469c742-a4e2-4026-843a-0c651ddd0cca.png)

**Comment by lbirkert** [comment]
> Select between visual styles

Will there be a fixed number of options that the user can choose from or will there be an component editor where the user can for example add a weather icon to his clock.

**Comment by lbirkert** [comment]
If so, we could also minimize this Problem by just creating more components, so that styles would be more about font weights, sizes, families and text color and not about variants in layout and positioning.

**Reply by Ari24-cb24** [reply]
Yeah, I think that it might go into this direction. When I started with the overhaul of the widget settings page I had a button for each widget in mind, which can add some css to the widget e.g. `box-shadow` (as a filter so it clips the text), `color` (maybe some sort of palette), `font-weight` and some more options (kind of like the widget add button, such that the widget has a pre-defined set of settings and can add "unlimited" options). For the clock, the idea is easily solvable by just adding an option to remove the date. 

Also, my original idea mainly came from JaxCore (basically Windows 7 Desktop Gadgets but better) as seen in the image. They do a lot more with styling and thus have pre-defined styles. This will also have its place in MyAnimeTab but with custom css instead (and of course some kind of online service for users to upload their snippets and download them in MyAnimeTab).
![grafik](https://user-images.githubusercontent.com/56089155/210968963-5bed4faa-3ff7-4d82-801a-cf3df9df4af6.png)

**Comment by lbirkert** [comment]
Mhmm yep, when there are a lot of widgets in dense space it'd probably also make sense to have a group widgets button which makes their relative position to another locked, so only the whole widget group may be moved, scaled or rotated accordingly.

**Comment by lbirkert** [comment]
There is basically always 2 routes to go to when designing customable interfaces: The iOS way: only making a few widget version that generally look good to the most people and fit very well with another, so that even users unexperienced users can make a consistent composition of widgets; servers a large portion of users, not all, or the linux way: allows for extreme customizability (more than most users need), which allows for the potential to serve all users. For most users this will be verbose tho, so perhaps a mix of these two would be the best?

**Reply by Ari24-cb24** [reply]
I agree with both. I like the "highly customizable" aspect which MyAnimeTab provides and will provide in the future but I also want something easy, something that doesn't take all too long to set up. I think having the abillity to add custom js and css to existing widgets and thus change them is a good idea. I also like the idea of having a community-library consisting of such skins and custom widgets (and maybe more at a later point of time)
